component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -2693,4 +2772,89 @@ private boolean isOTPMismatched(AuthenticationContext context) {
         return Boolean.parseBoolean(String.valueOf(
                 context.getProperty(EmailOTPAuthenticatorConstants.CODE_MISMATCH)));
     }
+
+    private boolean isIdfInitiatedFromEmailOTP() {
+
+        return Boolean.TRUE.equals(
+                authenticationContext.getProperty(EmailOTPAuthenticatorConstants.IS_IDF_INITIATED_FROM_AUTHENTICATOR));
+    }
+
+    private String validateIdentifierFromRequest(HttpServletRequest request)
+            throws AuthenticationFailedException {
+
+        String identifierFromRequest = request.getParameter(EmailOTPAuthenticatorConstants.USER_NAME);
+        if (StringUtils.isBlank(identifierFromRequest)) {
+            throw new InvalidCredentialsException(EmailOTPAuthErrorConstants.ErrorMessages.EMPTY_USERNAME.getCode(),
+                    EmailOTPAuthErrorConstants.ErrorMessages.EMPTY_USERNAME.getMessage());
+        }
+        return identifierFromRequest;
+    }
+
+    /**
+     * This method is used to resolve the user from authentication response from identifier handler.
+     *
+     * @param request The httpServletRequest.
+     * @param context The authentication context.
+     * @throws AuthenticationFailedException In occasions of failing.
+     */
+    public org.wso2.carbon.user.core.common.User resolveUser(HttpServletRequest request, AuthenticationContext context)
+            throws AuthenticationFailedException {
+
+        String username = validateIdentifierFromRequest(request);
+        validateEmailUsername(username, context);
+        username = FrameworkUtils.preprocessUsername(username, context);
+        org.wso2.carbon.user.core.common.User user = new org.wso2.carbon.user.core.common.User();
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        String tenantDomain = MultitenantUtils.getTenantDomain(username);
+
+        user.setUsername(tenantAwareUsername);
+        user.setUserStoreDomain(UserCoreUtil.extractDomainFromName(username));
+        user.setTenantDomain(tenantDomain);
+        return user;
+    }
+
+    private void setResolvedUserInContext(AuthenticationContext context, org.wso2.carbon.user.core.common.User user) {
+
+        Map<String, Object> authProperties = context.getProperties();
+        if (MapUtils.isEmpty(authProperties)) {
+            authProperties = new HashMap<>();
+            context.setProperties(authProperties);
+        }
+
+        String username = UserCoreUtil.addTenantDomainToEntry(user.getUsername(), user.getTenantDomain());
+        username = FrameworkUtils.prependUserStoreDomainToName(username);
```

Comment:
Should pass the user store name to prepend

Reply:
No. It is obtained from the threadLocal using the method `UserCoreUtil.getDomainFromThreadLocal()` from inside the `prependUserStoreDomainToName` method